### ISSUE TITLE
Form for adding creators to works or collections

### ISSUE THREAD

**Issue by awead** [psu-libraries/cho#733]
Builds the necessary html for entering creators and roles in a work or collection. This also includes the code required to validate the role with the Agent and return any errors if either is invalid. The Agent controlled vocabulary is already done, but the roles should be specified and validated here as MARC relators.

MARC relators can be a simple datalist pulled from http://id.loc.gov/vocabulary/relators.html. A json file might be best here. 

Child of #87 
Blocked by #732 
[Related: #191  NT]

**Comment by awead** [comment]
TODO
* create a validator that checks that the creator and roles exist
* use the field type from #732 to create the necessary UI elements to display dropdowns
* agents and roles HTML5 datalist elements
